### ISSUE TITLE
[Android] Cookie banner blocker

### ISSUE THREAD

**Issue by anthonypkeane** [brave/brave-browser#8974]
## Description 
Linked to https://github.com/brave/brave-browser/issues/5318

Please implement a cookie banner blocker on Android

cc @jamesmudgett or @karenkliu for design

**Comment by DragoCubed** [comment]
related to #9338

**Comment by kjozwiak** [comment]
The above requires `1.45.104` or higher for `1.45.x` verification 👍

**Comment by kjozwiak** [comment]
Labelling as `QA/Blocked` as we'll need to wait for https://github.com/brave/brave-browser/issues/26078 to be fixed and uplifted into `1.45.x`. We'll also need to enable `BraveAdblockCookieListOptInStudy` on `BETA` and `Release` before we can verify the above as we'll be enabling this feature via Grffin on both `Desktop` & `Android`.

@Uni-verse @srirambv I can run through this as I have the most context/been working on this for a while.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung S10+` running `Android 12` using the following build(s):

```
Brave | 1.45.111 Chromium: 107.0.5304.62 (Official Build) (32-bit)
--- | ---
Revision | 1eec40d3a5764881c92085aaee66d25075c159aa-refs/branch-heads/5304@{#942}
OS | Android 12; Build/SP1A.210812.016
```

* enabled the feature via `brave://flags#brave-adblock-cookie-list-opt-in`
* ensured that the `cookie consent` panel/notification appears after 5 refresh/loads
  * ensured that the `cookie consent` panel/notification appears when refreshing the same page five times
  * ensured that the `cookie consent` panel/notification appears when loading five different websites in new tabs
* ensured that clicking/tapping on `Block cookie notices` enables the feature and dismisses the panel/notification
  * ensured that `Easylist-Cookie List` is enabled via `brave://adblock`
  * ensured that `Block cookie consent notices` appears via `Hamburger Menu` -> `Brave Shields & privacy` and is enabled
  * ensured that the second image wasn't appearing via https://dev-pages.brave.software/filtering/cookie-banners.html

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------
![Screenshot_20221024-091505_Brave](https://user-images.githubusercontent.com/2602313/197537004-49487cdc-c820-433a-a82f-08e7ecb1d7a9.jpg) | ![Screenshot_20221024-092615_Brave](https://user-images.githubusercontent.com/2602313/197537030-7fc3997c-8c61-4d5d-badf-c67fd5224c50.jpg) | ![Screenshot_20221024-092629_Brave](https://user-images.githubusercontent.com/2602313/197537040-5ac12cf1-bda7-47f5-b8c8-d2822e1feae0.jpg) | ![Screenshot_20221024-092058_Brave](https://user-images.githubusercontent.com/2602313/197537046-67026937-fcb2-4472-bf5a-7d0079da771b.jpg)

* ensured that clicking/tapping on `No thanks` doesn't enable the feature and dismisses the panel/notification
  * ensured that `Easylist-Cookie List` is not checked off via `brave://adblock`
  * ensured that `Block cookie consent notices` appears via `Hamburger Menu` -> `Brave Shields & privacy` and is disabled
  * ensured that the second image wasn't appearing via https://dev-pages.brave.software/filtering/cookie-banners.html

**`Example`** | **`Example`** | **`Example`**
---------------|-----------------|---------------
![Screenshot_20221024-093213_Brave](https://user-images.githubusercontent.com/2602313/197539105-708e46cc-d52e-4544-b249-eeb29c7619d7.jpg) | ![Screenshot_20221024-093539_Brave](https://user-images.githubusercontent.com/2602313/197539144-2d3b80ea-5f2b-4c55-8b10-9263f8c16996.jpg) | ![Screenshot_20221024-093559_Brave](https://user-images.githubusercontent.com/2602313/197539200-2a0e0ddc-1c20-4f26-9aaf-2dce2d38dccd.jpg)

* ensured that you can scroll through a website without the `cookie consent` panel/notification dismissing
* ensured that tapping outside of the panel/notification didn't dismiss the panel/notification
* ensured that the `cookie consent` panel/notification is dismissed whem:
  * ensured panel/notification is being dismissed when opening a new tab
  * ensured panel/notification is being dismissed when closing active tab (where `cookie consent` panel is visible
  * ensured panel/notification is being dismissed when switching tabs
* ensured that enabling the feature via `Block cookie consent notices` settings works as expected if `No, thanks` was selected
  * ensured that the second image wasn't appearing via https://dev-pages.brave.software/filtering/cookie-banners.html
* ensured that once dismissed, the `cookie consent` panel/notification doesn't appear again
